### PR TITLE
Fix internal port configuration

### LINKED ISSUE #2484: Unable to show the Catalog tab
Just installed KubeApps on Tanzu Kubernetest cluster on vSphere with these steps:
1. helm repo add bitnami https://charts.bitnami.com/bitnami
2. kubectl create namespace kubeapps
3. helm install kubeapps --namespace kubeapps \
  --set assetsvc.service.port=9090 \
  --set frontend.service.type=LoadBalancer \
  --set useHelm3=true \
    bitnami/kubeapps
4. kubectl create --namespace default serviceaccount kubeapps-operator
5. kubectl create clusterrolebinding kubeapps-operator --clusterrole=cluster-admin --serviceaccount=default:kubeapps-operator
6. kubectl get secret $(kubectl get serviceaccount kubeapps-operator -o jsonpath='{range .secrets[*]}{.name}{"\n"}{end}' | grep kubeapps-operator-token) -o jsonpath='{.data.token}' -o go-template='{{.data.token | base64decode}}' && echo
7. login to web UI and click Catalog tab.
8. See an error message "An error occurred while fetching the catalog:"
kubeappserror

### PR DESCRIPTION
**Pull request by andresmgot**
### Description of the change

There were several issues if a user wanted to change an internal port (e.g. `assetsvc` port):

 - Several times, the port `8080` was hardcoded in the templates
 - The services require to set the env var `PORT` to change the port
 - (still an "issue") It's necessary to set the liveness and readiness probe port as well. (e.g. `--set assetsvc.service.port=9090 --set assetsvc.livenessProbe.httpGet.port=9090 --set assetsvc.readinessProbe.httpGet.port=9090`).

This PR fixes the first 2 items. Since there is no common use case for wanting to change the port, I haven't addressed the third.

I have also changed the documentation to avoid confussion or people trying incomplete commands.

### Applicable issues

  - fixes #2484

### CODE REVIEW

**Review by antgamdia — APPROVED** [review]
Nice! While it is not that critical, it is great to provide a configurable way to change the hardcoded values. Thanks!

File: chart/kubeapps/README.md [hunk 1]
```diff
@@ -57,11 +57,11 @@ Specify each parameter using the `--set key=value[,key=value]` argument to `helm
 
 ```bash
 helm install kubeapps --namespace kubeapps \
-  --set assetsvc.service.port=9090 \
+  --set ingress.enabled=true \
```
Comment (absoludity) [inline chart/kubeapps/README.md:60]: Heh, much better example now :)

**Review by absoludity — APPROVED** [review]
> This PR fixes the first 2 items. Since there is no common use case for wanting to change the port, I haven't addressed the third.

I don't understand - if we have a value for `assetsvc.service.port`, then shouldn't the liveness probe use the same value? Why does the chart have a separate `assetsvc.livenessProbe.httpGet.port` option? Or at least, why are we setting that with a default in the values.yaml rather than defaulting to `assetsvc.service.port` ? (same for other deploys). It looks like the only reason may be that we're using `toYaml` to specify the probes from the values in the deployments (probably because that's a bitnami chart thing?).

If we switched the `*.livenessProbe.httpGet.port`'s to all not have a default, we could expand the probes in the deployment container specs and use the value for the `*.livenessProbe.httpGet.port` if set, otherwise default to the port of the actual service in our case?

Anyway, not required for this one, just mentioning in case you think it's worth doing for a follow-up. Currently people can just set the probes explicitly as well if needed.

**Comment by andresmgot** [comment]
> > This PR fixes the first 2 items. Since there is no common use case for wanting to change the port, I haven't addressed the third.
> 
> I don't understand - if we have a value for `assetsvc.service.port`, then shouldn't the liveness probe use the same value? Why does the chart have a separate `assetsvc.livenessProbe.httpGet.port` option? Or at least, why are we setting that with a default in the values.yaml rather than defaulting to `assetsvc.service.port` ? (same for other deploys). It looks like the only reason may be that we're using `toYaml` to specify the probes from the values in the deployments (probably because that's a bitnami chart thing?).
> 
> If we switched the `*.livenessProbe.httpGet.port`'s to all not have a default, we could expand the probes in the deployment container specs and use the value for the `*.livenessProbe.httpGet.port` if set, otherwise default to the port of the actual service in our case?
> 
> Anyway, not required for this one, just mentioning in case you think it's worth doing for a follow-up. Currently people can just set the probes explicitly as well if needed.

Yes, you are right, that would be the implementation I just didn't take the time to update all the probes that we have in the chart because there is no real use-case for changing the port (AFAIK).

**Pull request merged**